doc/aaa/TACACS+ Requirement.md

```diff
@@ -0,0 +1,161 @@
+	- All privilege level share same whitelist to simplify design.
+	- Different privilege level have different permission to run these command.
+	- All commands in sudoers will add to the whitelist. and sudoers config file still need for RO users, this is because when remote TACACS server not avaliable, we need use local group permission for failover.
+
+- Disable user behavior in shell:
+	- Changing directories with the cd builtin.
+		- User still can access files in other folder, and use 'ls' command to list content in other folder, for example:
+        ```
+            test@testhost:~$ ls -l /etc/
+            total 880
+            drwxr-xr-x 3 test test       4096 Aug 22  2020 NetworkManager
+            drwxr-xr-x 7 test test       4096 Aug 22  2020 X11
+            drwxr-xr-x 3 test test       4096 Aug 22  2020 acpi
+        ```
+	- Setting or unsetting the values of the SHELL, PATH, HISTFILE, ENV, or BASH_ENV variables.
+	- Specifying command names containing slashes.
+	- Importing function definitions from the shell environment at startup.
+	- Parsing the value of SHELLOPTS from the shell environment at startup.
+	- Redirecting output using the ‘>’, ‘>|’, ‘<>’, ‘>&’, ‘&>’, and ‘>>’ redirection operators.
+	- Builtin commands:
+
+| **Command** | **Behavior**                |
+| -------- | -------------------------- |
+| . | Specifying a filename containing a slash as an argument.     |
+| history     | Specifying a filename containing a slash as an argument. |
+| hash     | Specifying a filename containing a slash as an argument to the -p option. |
+| exec     | Specifying a filename containing a slash as an argument to the -p option. |
+| deleting/adding     | Use the -f and -d options to the enable builtin. |
+| enable     | Using the 'enable' builtin command to enable disabled shell builtins. |
+| command     | Specifying the -p option to the 'command' builtin command. |
+| set     | Turning off restricted mode with ‘set +r’ or ‘set +o restricted’. |
+
+	- All these behavior disabled only for user input, command in script will not be affected, see here for more details: https://www.gnu.org/software/bash/manual/html_node/The-Restricted-Shell.html
+
+
+- Supported Authorization types:
+	- EXEC: user session authorization support. this happen when user login.
+	- Command: user run command in shell.
+
+- Failover:
+	- Authorization will happen before execute, if remote TACACS server not available, use local group based authorization as failover.
+
+
+## 1.3 Accounting
+ - Accounting when:
+ 	- User login to SONiC host.
+ 	- User run command on host: 
+		- Command start run.
+		- Command finish.
+		- For recursive command/script, only the top level command have Accounting.
+- Failover:
+	- Use syslog as backup when remote TACACS not avaliable.
+
+
+## 1.4 User script
+ - Any script in whitelist can run with Authorization and Accounting. 
+ - If user create a script, admin user can use config command add script to whitelist.
```

Comment:
>whitelist [](http://example.com/codeflow?start=76&length=9)

Is it true that tacacs server owns this whitelist? How can a sonic change the server's whitelist?

Reply:
Will add example to document.

Tacacs server not have this whitelist, and sonic can't change the tacacs server setting.
However, for user script, if the script name follow some naming convention, and tacacs server side allow run those script which follow the naming convention, user can create their own script, for example:

1. Tacacs service allow RW user run any script named as 'user_script_*'
2. RW user create a new script on sonic host, script name is 'user_script_collect_information.sh'
3. Then user can add this script to whitelist and run it.

But if tacacs service not have any config, then user can't run their script because Authorization will block it.